pkg/controller/compliancesuite/compliancesuite_controller.go

```diff
@@ -590,13 +592,14 @@ func (r *ReconcileComplianceSuite) applyMcfgRemediationAndPausePool(rem compv1al
 	return nil
 }
 
-func (r *ReconcileComplianceSuite) getAffectedMcfgPool(scan *compv1alpha1.ComplianceScan, mcfgpools *mcfgv1.MachineConfigPoolList) (mcfgv1.MachineConfigPool, bool) {
-	for _, pool := range mcfgpools.Items {
-		if utils.McfgPoolLabelMatches(scan.Spec.NodeSelector, &pool) {
+func (r *ReconcileComplianceSuite) getAffectedMcfgPool(scan *compv1alpha1.ComplianceScan, mcfgpools *mcfgv1.MachineConfigPoolList) (*mcfgv1.MachineConfigPool, bool) {
```

Comment:
Since we're already changing the signature of this function to (ptr, bool) why not drop the bool and check if the pointer is non-nil instead as a flag if the pool exists?

Reply:
done